Add cluster migration for new columns

charmik-redhat opened the pull request:
### Description

<!-- A detailed explanation of the changes in your PR. Feel free to remove this section if the title of your PR is sufficiently descriptive. -->

New search tags were added in #10844 . As that created new columns, a db migration was added to populate those new columns. However, since clusters are updated at startup, that would populate the new columns too. So we decided to remove the migration [here](https://github.com/stackrox/stackrox/pull/10844/commits/68b6ff67aa94423846beb87f89c1516189502d2d#diff-f034035c383d348e1cafc573ebe597539752cc3b3edad9e05b77aa53391907e7) . Upon further testing, we realized that degraded and uninitialized clusters do not get updated during startup. So we need the migration to update those clusters. Otherwise filtering by new columns would not work for those clusters. This PR adds back the removed migration.

### User-facing documentation

(*must be* 2 items and both *must be* checked)
<!-- Remove conflicting items that won't be checked. -->

- [X] CHANGELOG update is not needed
- [X] Documentation is not needed

### Testing

- [X] inspected CI results

#### Automated testing

(*must be* at least 1 item and all items *must be* checked)
<!-- Remove item(s) that don't apply and won't be checked. -->

- [X] added unit tests
  <!-- Please explain why unless it's obvious, e.g., the PR is a one-line comment change. -->

#### How I validated my change

- Unit tests
- Manual test to verify both healthy and degraded clusters are migrated correctly

<!--
Use this space to explain **how you validated** that **your change functions exactly how you expect it**.
Feel free to attach JSON snippets, curl commands, screenshots, etc. Apply a simple benchmark: would the information you
provided convince any reviewer or any external reader that you did enough to validate your change.

It is acceptable to assume trust and keep this section light, e.g. as a bullet-point list.

It is acceptable to skip testing in cases when CI is sufficient, or it's a markdown or code comment change only.
It is also acceptable to skip testing for changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting, fixing, etc. Make sure you validate the change
ASAP after it gets merged or explain in PR when the validation will be performed.
Explain here why you skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.
-->

change me!

clickboo reviewed (approved):
code looks good, but @ksurabhi91  has another migration outstanding for policy exclusions so one of you needs to update the numbering

charmik-redhat commented:
> code looks good, but @ksurabhi91 has another migration outstanding for policy exclusions so one of you needs to update the numbering

Rebased and updated it

dashrews78 commented on migrator/migrations/m_204_to_m_205_clusters_platform_type_and_k8_version/migration.go:
```diff
@@ -0,0 +1,24 @@
+// Code generated by make bootstrap_migration generator. DO NOT EDIT.
+package m203tom204
```
package didn't get updated.
